onnxruntime/core/providers/cuda/reduction/reduction_ops.cc

```diff
@@ -748,7 +748,7 @@ Status ReduceKernel<allow_multi_axes>::ComputeImpl(OpKernelContext* ctx, cudnnRe
     IAllocatorUniquePtr<float> temp_X = GetScratchBuffer<float>(input_count);                                        \
     Impl_Cast<CudaT, float>(Stream(), reinterpret_cast<const CudaT*>(X->template Data<T>()), temp_X.get(), X->Shape().Size()); \
                                                                                                                      \
-    ORT_RETURN_IF_ERROR(reduce_desc.Set(cudnn_reduce_op, cudnn_type_X, CUDNN_REDUCE_TENSOR_FLATTENED_INDICES));      \
+    ORT_RETURN_IF_ERROR(reduce_desc.Set(cudnn_reduce_op, cudnn_type_X, CUDNN_REDUCE_TENSOR_NO_INDICES));             \
```

Comment:
@iK1D - can you verify this change is okay?

This code is in a template specialization for `CUDNN_REDUCE_TENSOR_NO_INDICES` but it was using `CUDNN_REDUCE_TENSOR_FLATTENED_INDICES` here.  I believe it should be using `CUDNN_REDUCE_TENSOR_NO_INDICES`.

Reply:
@jessebenson, I am actually not quite sure which one is correct, I am not that familiar with CUDNN. Previously in this file there are three pieces of same code for these three int types, and all of them are using CUDNN_REDUCE_TENSOR_FLATTENED_INDICES (as in https://github.com/microsoft/onnxruntime/blob/4c6118eb491ce09d67ef8d3117ee623974b75f8a/onnxruntime/core/providers/cuda/reduction/reduction_ops.cc#L740). What I did is just to make them into one to save duplicated code.

@weixingzhang, could you please help to answer Jesse's question here?

Reply:
Yeah, Jesse is right. It should be CUDNN_REDUCE_TENSOR_NO_INDICES since the code is specialized for CUDNN_REDUCE_TENSOR_NO_INDICES.

Reply:
@iK1D - can you Approve?  Looks like I made you a Required approver to merge :)